games/rpg/src/lib.rs

```diff
@@ -0,0 +1,1483 @@
+            let key_bucket: Bucket = self.system_vault.take(1);
+            // Data from Proofs
+            let mut nft_data: Account = nft_proof.non_fungible().data();
+            let weapon_data: Weapon = gear.non_fungibles()[0].data();
+            let helmet_data: Armor = gear.non_fungibles()[1].data();
+            let chest_data: Armor = gear.non_fungibles()[2].data();
+            let pants_data: Armor = gear.non_fungibles()[3].data();
+            let gloves_data: Accessory = gear.non_fungibles()[4].data();
+            let belt_data: Accessory = gear.non_fungibles()[5].data();
+            let shoes_data: Accessory = gear.non_fungibles()[6].data();
+            // Getting data of selected stage:
+            let data = self.stage_data.get(&stage).unwrap();
+            let enemy_1_hp: Decimal = data[1].into();
+            let enemy_1_dmg: Decimal = data[2].into();
+            let enemy_1_def: Decimal = data[3].into();
+            let enemy_1_spd: Decimal = data[4].into();
+            let enemy_2_hp: Decimal = data[5].into();
+            let enemy_2_dmg: Decimal = data[6].into();
+            let enemy_2_def: Decimal = data[7].into();
+            let enemy_2_spd: Decimal = data[8].into();
+            let enemy_3_hp: Decimal = data[9].into();
+            let enemy_3_dmg: Decimal = data[10].into();
+            let enemy_3_def: Decimal = data[11].into();
+            let enemy_3_spd: Decimal = data[12].into();
+            let reward1_d_1 = data[13];
+            let reward2_d_1 = data[14];
+            let reward3_d_1 = data[15];
+            let exp_enemy_1: u128 = data[16].into();
+            let reward1_d_2 = data[17];
+            let reward2_d_2 = data[18];
+            let reward3_d_2 = data[19];
+            let exp_enemy_2: u128 = data[20].into();
+            let reward1_d_3 = data[21];
+            let reward2_d_3 = data[22];
+            let reward3_d_3 = data[23];
+            let reward1_w = data[24];
+            let reward2_w = data[25];
+            let reward3_w = data[26];
+            let exp_enemy_3: u128 = data[27].into();
+            // Assertions so no cheating
+            assert!(nft_proof.amount() == dec!("1"));
+            assert!(nft_proof.resource_address() == self.character_nft,);
+            assert!(nft_data.stage >= stage.into() || nft_data.stage == stage.into());
+            assert!(gear.resource_address() == self.item_nft,);
+            assert!(helmet_data.part == String::from("Helmet"));
+            assert!(chest_data.part == String::from("Chest"));
+            assert!(pants_data.part == String::from("Pants"));
+            assert!(gloves_data.part == String::from("Gloves"));
+            assert!(belt_data.part == String::from("Belt"));
+            assert!(shoes_data.part == String::from("Shoes"));
+            // Speed = sum of character + gear speed/Speed penality (Tier1 gear gives 1% penalty per item for 6% total penalty)
+            let speed =
```

Comment:
I'm not entirely certain as to what this does. However, can't the math here be simplified to: `(nft_data.speed + gloves_data.speed + belt_data.speed + shoes_data.speed)(1 - helmet_data.weight - chest_data.weight - pants_data.weight - gloves_data.weight - belt_data.weight - shoes_data.weight)`? 

If so, can the math for other operations be simplified as well?